packages/frontend/src/view/components/NetworkSelector.tsx

```diff
@@ -9,32 +20,103 @@ interface Props {
 }
 
 export function NetworkDropdownSelector(props: Props) {
-  const options = props.chainsToDisplay.map((chainToDisplay) => ({
-    label: ChainId.getName(chainToDisplay),
-    value: ChainId.getName(chainToDisplay),
-  }))
+  // We want almost instance user feedback, but api call
+  // and refresh itself will take a bit thus affecting ux
+  const [optimisticSelect, setOptimisticSelect] = useState<ChainId>(
+    props.chainId,
+  )
 
-  function onChange(option: DropdownOption) {
-    const chainId = ChainId.fromName(option.value)
+  function setChainId(chainId: ChainId) {
+    setOptimisticSelect(chainId)
     props.setChainId(chainId)
   }
 
-  const defaultValue = options.find(
-    (option) => option.value === ChainId.getName(props.chainId),
-  )
-
   if (!props.chainsToDisplay.includes(props.chainId)) {
-    props.setChainId(props.chainsToDisplay[0])
+    setChainId(props.chainsToDisplay[0])
   }
 
+  const pills = props.chainsToDisplay
+    .map((chainToDisplay, i) => ({
+      key: i,
+      icon: getIconForChain(chainToDisplay),
+      label: getPrettyChainName(chainToDisplay),
+      isActive: chainToDisplay === optimisticSelect,
+      onClick: () => setChainId(chainToDisplay),
+    }))
+    .sort((a, b) => a.label.localeCompare(b.label))
+    .map((props) => <PillSelector {...props} />)
+
   return (
-    <section className="bg-gray-900 px-6 pb-0 pt-6">
-      <label className="mb-4 text-xs text-gray-500">Select network</label>
-      <Dropdown
-        defaultValue={defaultValue}
-        onChange={onChange}
-        options={options}
-      />
+    <section className="bg-gray-500 px-6 py-8 md:justify-center">
+      <div className="mx-auto max-w-fit">
+        <span className="text-xs text-gray-15">Select network</span>
+        <div className="scrollbar-h-1.5 flex items-stretch gap-3 overflow-x-auto py-2 scrollbar scrollbar-track-gray-50 scrollbar-thumb-[#eef36a]">
+          {pills}
+        </div>
+      </div>
     </section>
   )
 }
+
+function PillSelector({
+  icon,
+  label,
+  isActive,
+  onClick,
+}: {
+  icon: React.ReactNode
+  label: string
+  isActive: boolean
+  onClick: () => void
+}) {
+  const focusRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (focusRef.current && isActive) {
+      focusRef.current.scrollIntoView({ behavior: 'smooth', inline: 'center' })
+    }
+  }, [focusRef, isActive])
+
+  return (
+    <div
+      ref={focusRef}
+      className={cx(
+        'flex min-w-fit cursor-pointer items-center justify-center gap-1 rounded py-3 pl-4 pr-6 text-center',
+        isActive
+          ? 'bg-yellow-100 text-black'
+          : 'bg-gray-300 text-white brightness-100 filter transition-all duration-300 hover:brightness-[130%]',
+      )}
+      onClick={onClick}
+    >
+      <div className="flex h-[30px] w-[30px] items-center justify-center">
```

Comment:
Let's make the icon a bit smaller, similar to the designs
```suggestion
      <div className="flex h-6 w-6 items-center justify-center">
```

Reply:
[feat: unify paddings across netowork pills](https://github.com/l2beat/lz-monitoring/pull/66/commits/b5e5778f86c4848569c84001a66a332050f9db31)